nncf/common/quantization/quantizer_propagation/solver.py

```diff
@@ -195,6 +199,129 @@ def is_final_qconfig_compatible_to_initial(initial_qconfig: QuantizerConfig):
+                return True
+        return False
+
+    def _update_traverse_path_flags(self, node_metatype: OperatorMetatype) -> None:
+        if node_metatype in self._post_processing_marker_metatypes:
+            self._post_processing_marker_encountered = True
+
+    def _get_node_metatype(self, node_key: str) -> OperatorMetatype:
+        node = self._quant_prop_graph.nodes[node_key]
+        return node.get(self._quant_prop_graph.OPERATOR_METATYPE_NODE_ATTR)
+
+    def _is_node_operator(self, node_key: str) -> bool:
+        node = self._quant_prop_graph.nodes[node_key]
+        return node.get(self._quant_prop_graph.NODE_TYPE_NODE_ATTR) == QuantizerPropagationStateGraphNodeType.OPERATOR
+
+    def _get_ignored_node_keys(self, node_keys: List[str]) -> List[str]:
+        output = []
+        for node_key, node_metatype in zip(node_keys, map(self._get_node_metatype, node_keys)):
+            if node_metatype not in self._post_processing_marker_metatypes:
+                output.append(node_key)
+        return output
+
+    def get_post_processing_node_keys(self) -> List[str]:
+        """
+        Finds out the nodes of the QuantizerPropagationStateGraph, which are in post-processing part of the model.
+        Starting from the output nodes all the nodes are added, until the quantizable nodes with weights are faced.
+        If the path with the nodes has the post-processing marker node,
+        all the nodes in this path will be added into ignored.
+        :return: The list of the node keys to be ignored.
+        """
+
+        visited_nodes = set()
+
+        def backward_traverse_function(node_key: str, output: List[str],
+                                       visited_nodes: Set[str]) -> Tuple[bool, List[str]]:
+            """
+            Realizes the search of the quantization ignored nodes in graph.
+            Only QuantizerPropagationStateGraphNodeType.OPERATOR nodes are processed during the traversing.
+            If the current node is in the list of the quantizable nodes with weights,
+             the traversing is being stopped.
+            The new forward traversing from the current node starts.
+            If the quantizable nodes with weights is faced in forward traversing faced,
+             the original backward traversing is being stopped.
+
+            :param node_key: node key to check, whether the traversing has to be stopped or not
+            and whether the node should be added to the traversed path.
+            :param output: Path contains the list of the visited nodes.
+            :param visited_nodes: Set stores whether the particular node was visited before or not.
+            :return: The first value shows whether the traversing finished,
+            the second one is traversing path containing the visited nodes.
+            """
+
+            def forward_traverse_function(node_key: str, are_weight_nodes: List[bool], visited_nodes: Set[str]) \
```

Comment:
Are you even using the `are_weight_nodes` parameter? If not, remove it.

Reply:
Change from `are_weight_nodes: List[bool]` to `output: List[str]`